src/Selenium2Driver.php

```diff
@@ -946,50 +862,68 @@ public function evaluateScript($script)
             $script = 'return ' . $script;
         }
 
-        return $this->wdSession->execute(array('script' => $script, 'args' => array()));
+        return $this->webDriver->executeScript($script);
     }
 
     /**
-     * {@inheritdoc}
+     * Waits some time or until JS condition turns true.
+     *
+     * @param int                                        $timeout   timeout in milliseconds
+     * @param WebDriverExpectedCondition|\Closure|string $condition JS condition / Closure
+     *
+     * @return bool
      */
     public function wait($timeout, $condition)
     {
-        $script = "return $condition;";
-        $start = microtime(true);
-        $end = $start + $timeout / 1000.0;
+        $seconds = $timeout / 1000.0;
+        $wait = $this->webDriver->wait($seconds);
 
-        do {
-            $result = $this->wdSession->execute(array('script' => $script, 'args' => array()));
-            usleep(100000);
-        } while (microtime(true) < $end && !$result);
+        if (is_string($condition)) {
+            $script = "return $condition;";
+            $condition = function (RemoteWebDriver $driver) use ($script) {
+                return $driver->executeScript($script);
+            };
+        }
 
-        return (bool) $result;
+        try {
+            return (bool) $wait->until($condition);
+        } catch (TimeOutException $e) {
+            return false;
+        }
     }
 
     /**
      * {@inheritdoc}
      */
     public function resizeWindow($width, $height, $name = null)
     {
-        $this->wdSession->window($name ? $name : 'current')->postSize(
-            array('width' => $width, 'height' => $height)
-        );
+        $dimension = new WebDriverDimension($width, $height);
+        if ($name) {
+            throw new \Exception('Named windows are not supported yet');
+        }
+
+        $this->webDriver->manage()->window()->setSize($dimension);
     }
 
     /**
      * {@inheritdoc}
      */
     public function submitForm($xpath)
     {
-        $this->findElement($xpath)->submit();
+        $element = $this->findElement($xpath);
+        $element->submit();
     }
 
     /**
      * {@inheritdoc}
      */
     public function maximizeWindow($name = null)
     {
-        $this->wdSession->window($name ? $name : 'current')->maximize();
+        if ($name) {
+            throw new \Exception('Named window is not supported');
```

Comment:
should be an `UnsupportedDriverActionException` instead

Reply:
done